parasol/solr/client.py

```diff
@@ -0,0 +1,131 @@
+import inspect
+import json
+import logging
+import sys
+import time
+from urllib.parse import urljoin
+
+from attrdict import AttrDict
+import requests
+
+# Start making Django agnostic from the get-go
+try:
+    from django.settings import settings
+    from django.core.exceptions import ImproperlyConfigured
+except ImportError:
+    pass
+
+
+from parasol import __version__ as parasol_version
+from parasol.solr.schema import Schema
+from parasol.solr.update import Update
+from parasol.solr.admin import CoreAdmin
+
+logger = logging.getLogger(__name__)
+
+
+
+class SolrClientException(Exception):
+    '''Base class for all exceptions in this module'''
+    pass
+
+
+class CoreForTestExists(SolrClientException):
+    '''Raised when default core for running unit tests exists'''
+
+
+
+class SolrClient:
+    '''Base class for all SolrClient with sane development defaults'''
+    #: Url for solr base instance
+    solr_url = 'http://localhost:8983/solr'
+    #: core admin handler
+    core_admin_handler = 'admin/cores'
+    #: select handler
+    select_handler = 'select'
+    #: schema handler
+    schema_handler = 'schema'
+    # update handler
+    update_handler = 'update'
+    #: core or collection
+    collection = ''
+    # commitWithin definition
+    commit_within = 1000
+
+    def __init__(self, *args, **kwargs):
+        # For now, a generous init that will override valuesbut not
+        # choke on an unexpected one.
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+        # Don't allow session configuration to be so easily overwritten
+        self.session = requests.Session()
+        self.session.headers = {
+            'User-Agent': 'Parasol/%s' % parasol_version
```

Comment:
Might be a good idea to include the requests version here also, just in case we ever run into behavior issues that are specific to versions of request. I checked the docs and don't see an easy way to include it automatically, but not hard to import and include. Maybe `parasol/version (python-requests/version)`.

I prefer lowercase parasol.

Reply:
Sounds good to me. I've added this in a commit I'll push.